src/main/webapp/resources/js/apis/cart/cart.js

```diff
@@ -1,47 +1,49 @@
 import axios from "axios";
 import { setBaseUrl } from "../../utilities/url-utilities";
 
-const url = setBaseUrl(`cart`);
+const AJAX_URL = setBaseUrl(`/ajax/cart`);
 
 /**
  * Add samples for a project to the cart.
  * @param {number} projectId Identifier for the project the samples are from.
  * @param {array} samples array of sample {ids } to add to cart.
- * @returns {AxiosPromise<any>}
+ * @returns {Promise<{count: any}>}
  */
 export const putSampleInCart = async (projectId, samples) =>
-  axios.put(url, {
-    projectId,
-    samples
-  });
+  axios
+    .post(AJAX_URL, {
+      projectId,
+      sampleIds: samples.map((s) => s.id),
+    })
+    .then(({ data }) => data);
 
 /**
  * Get the current number of samples in the cart
- * @returns {Promise<void>}
+ * @returns {Promise<{count: any}>}
  */
 export const getCartCount = async () => {
-  return axios.get(`${url}/count`).then(response => ({ count: response.data }));
+  return axios.get(`${AJAX_URL}/count`).then(({ data }) => ({ count: data }));
 };
 
 /**
  * Get the current state of the cart.
  * @returns {Promise<void | never>}
  */
 export const getCart = async () =>
-  axios.get(`${url}`).then(response => response.data);
+  axios.get(`${AJAX_URL}`).then((response) => response.data);
 
 export const getCartIds = async () =>
```

Comment:
Add comments to functions

Reply:
Added in edb5a72614c8656f5bef85e0f90bce70295b0a90